### DISCUSSION TITLE
Populate Event Header with Dynamics Key (ie ftpUser or other Info)

### DISCUSSION THREAD

**Discussion by idachour59** [drakkan/sftpgo#978] [Q&A]
Hello

I use sftpgo and EventManager that is really usefull for my activity.
I want to generate Event and  populate  my Event header with rooting Key that i want to use for my ftpUser. 
It will be root by the exhange to a specific queue.

Do you know if it is possible ?

**Comment by drakkan** [comment]
Sorry I didn't understand what you mean. Please explain with an example.

You can use the supported placeholders, take a look at the [docs](https://github.com/drakkan/sftpgo/blob/main/docs/eventmanager.md).

Also read the [support policy](https://github.com/drakkan/sftpgo#support-policy), thanks

**Reply by idachour59** [reply]
Sure and thanks for you reply.

right now, i receive from sftpgo this event

{
   "timestamp":"2022-07-15T07:22:46.332733411Z",
   "action":"upload",
   "username":"XYZ",
   "fs_path":"Path2MyFile/File.dat,
   "virtual_path":"File.dat",
   "file_size":913460,
   "status":1,
   "protocol":"SFTP",
   "ip":"00.11.22.33.44",
   "session_id":"SFTP_f1d8a243d73751a14048a06f69227303f9a62bf774917c0bdf07b64385d28a84_1",
   "
  }
This is the Payload generated by sftpgo when a user upload a File.
With RabbitMQ we can add some other metadata in the header of the message : example the User in order to create routing rules in RabbitMQ. 
I want to use the "Routing key" Metadata in the header in order to have
"Routing key":"USER"
 RabbitMQ will read the "Routing key" and forward to the dedicated queue.

**Comment by drakkan** [comment]
This should work if you switch to the development version and use the new event manager added in the web admin UI which allow to customize the events payload

**Reply by idachour59** [reply]
hi,

I have installed the dev version and try to use EvenManager but i have an issue after setting  rules and action for evenManager.
![image](https://user-images.githubusercontent.com/112820161/189522654-302c859b-7157-41ee-af55-463c00e68d4c.png)

I have the following  error in log file

`{"level":"info","time":"2022-09-11T10:26:33.060","sender":"Upload","local_addr":"10.2.4.56:2022","remote_addr":"10.2.5.0:49577","elapsed_ms":535,"size_bytes":913460,"username":"FRA11111100100010","file_path":"/srv/sftpgo/data/fic/in/FRA11111100100010_RES/FRA37967359300024_E_20220102_71.dat","connection_id":"SFTP_d3bda3630a6bb58e9b5099846e15b5c8daae1585e59fb73221e7bc1efa8f578f_1","protocol":"SFTP"}
{"level":"debug","time":"2022-09-11T10:26:33.060","sender":"SFTP","connection_id":"SFTP_d3bda3630a6bb58e9b5099846e15b5c8daae1585e59fb73221e7bc1efa8f578f_1","message":"transfer removed, id: 1 active transfers: 0"}
{"level":"debug","time":"2022-09-11T10:26:33.119","sender":"eventmanager","message":"executed command \"/etc/sftpgo/plugin/sftpgo-plugin-pubsub-linux-amd64\", elapsed: 59.206689ms, error: exit status 1"}
{"level":"error","time":"2022-09-11T10:26:33.119","sender":"eventmanager","message":"unable to execute action \"Send2_FRA11111100100010\" for rule \"FRA11111100100010\", elapsed 59.292572ms, err: action \"Send2_FRA11111100100010\" failed: exit status 1"}`

when i connect to the Pod and execute the command :
/etc/sftpgo/plugin/sftpgo-plugin-pubsub-linux-amd64 rabbit://sftpgo   it run because 

`root@sftpgo2-79d5ccdbc7-t52lk:/var/lib/sftpgo# /etc/sftpgo/plugin/sftpgo-plugin-pubsub-linux-amd64 rabbit://sftpgo

- [INFO]  starting sftpgo-plugin-pubsub: version=1.0.3-b7fc623-2022-06-04T08:11:08Z topic=rabbit://sftpgo instance id=""
- This binary is a plugin. These are not meant to be executed directly.
- Please execute the program that consumes these plugins, which will
- load any plugins automatically

root@sftpgo2-79d5ccdbc7-t52lk:/var/lib/sftpgo# `

does the plugin section in the sftpgo.json file need to be filled in to send notifications? 

Any idea ?

**Comment by idachour59** [comment]
this new feature seem to be very usefull but i think i use it with the wrong way.
Could you help me, please ?

- do I have to maintain the event generation configuration in the sftpgo.json?
- do i Keep the sftpgo-plugin-pubsub-linux-amd64 file that is used for send event in the last version ?
- which arguments should I enter in the argument field of the eventmanager

Thanks for your reply !
